### PR TITLE
Add env for switching gpt models

### LINKED ISSUE #41: [Feature]: Add support to switch between GPT models
### Feature Description

For instance, 3.5 turbo and 4. This should probably be handled via an env of some sort.

It’s possible there may need to be different prompts between models depending on specificity, too. The prompts I’ve made so far work best with 4.

### PR DESCRIPTION
**Pull request by mdrokz**
## Issue

- I added env to switch gpt models if the user doesnt provide any env by default it will use gpt-4 model
- I added OpenAI config info to the logger

Resolves #41 

## Testing Steps

1. I tested with gpt-3.5 turbo with no changes to the prompt it worked properly and fetched all the monster,habitat & dates info

### CODE REVIEW

File: src/utils/chat-gpt.ts [hunk 1]
```diff
@@ -18,13 +18,22 @@ async function askGPTChat(messages, debug) {
...
+                }),
+            );
         }
 
         const openai = new OpenAI({
             apiKey: process.env.OPENAI_API_KEY,
         });
         const chatCompletion = await openai.chat.completions.create({
-            model: 'gpt-4',
+            model: process.env.GPT_MODEL ?? 'gpt-4',
```
Comment (seriouslysean) [inline src/utils/chat-gpt.ts]: Move this to a variable higher in the file.

File: src/utils/chat-gpt.ts [hunk 2]
```diff
@@ -18,13 +18,22 @@ async function askGPTChat(messages, debug) {
             logger.info('-----Question-----');
             logger.info(stringifyJSON(messages));
             logger.info('');
+            logger.info('-----OpenAI Config-----');
+            logger.info(
+                stringifyJSON({
+                    model: process.env.GPT_MODEL ?? 'gpt-4',
+                    // Adjust this value (0 to 1) to control randomness. Lower values make the output more deterministic.
+                    // https://platform.openai.com/docs/guides/gpt/how-should-i-set-the-temperature-parameter
+                    temperature: 0.2,
```
Comment (seriouslysean) [inline src/utils/chat-gpt.ts]: Move this to a variable or config higher in the file that can be shared between functions.

File: src/utils/chat-gpt.ts [hunk 3]
```diff
@@ -18,13 +18,22 @@ async function askGPTChat(messages, debug) {
             logger.info('-----Question-----');
             logger.info(stringifyJSON(messages));
             logger.info('');
+            logger.info('-----OpenAI Config-----');
+            logger.info(
+                stringifyJSON({
+                    model: process.env.GPT_MODEL ?? 'gpt-4',
```
Comment (seriouslysean) [inline src/utils/chat-gpt.ts]: Make sure to add an entry in the env sample file. Also change the name to `OPENAI_GPT_MODEL`.

**Review by seriouslysean — CHANGES_REQUESTED** [review]
This is really close, couple small suggestions. I also wonder if there’s a better way to switch between models. Right now it’s largely manual but we only really support 3.5 turbo and 4.

Otherwise this is really close! Thanks so much for all the work on this.

**Comment by mdrokz** [comment]
> This is really close, couple small suggestions. I also wonder if there’s a better way to switch between models. Right now it’s largely manual but we only really support 3.5 turbo and 4.
> 
> Otherwise this is really close! Thanks so much for all the work on this.

We can also provide cmd arguments if env file is not present, so people can pass a cmd arg like this `---model gpt-3.5` or `--model gpt-4` like this. If this sounds good to you i could include it in this PR in the meantime i will work on the suggestions.

File: src/utils/chat-gpt.ts (hunk 3 above)
Comment (mdrokz) [reply src/utils/chat-gpt.ts]: > Make sure to add an entry in the env sample file. Also change the name to `OPENAI_GPT_MODEL`.

done

**Comment by mdrokz** [comment]
Hey i just pushed changes for the above suggestions

**Comment by seriouslysean** [comment]
I don’t think we need the arguments just now, only the env switch. Good future suggestion potentially, though.

File: .env.example [hunk 4]
```diff
@@ -2,3 +2,6 @@
 OPENAI_API_KEY=""
 # https://rollbar.com/signup/
 ROLLBAR_ACCESS_TOKEN=""
+
+# gpt-3.5-turbo or gpt-4
+OPENAI_GPT_MODEL=""
```
Comment (seriouslysean) [inline .env.example]: This isn’t a secret, default it to 4.
 Also add a return at the end of the file.

Comment (mdrokz) [reply .env.example]: > This isn’t a secret, default it to 4. Also add a return at the end of the file.

done

**Review by seriouslysean — DISMISSED** [review]
This is dope, thank you so much for the contribution!

**Comment by seriouslysean** [comment]
@mdrokz just fix the last couple lining issues and I’ll merge it!

**Comment by mdrokz** [comment]
> @mdrokz just fix the last couple lining issues and I’ll merge it!

Do you mean the linting issues? İ see the GitHub check is failing

**Comment by seriouslysean** [comment]
@mdrokz yeah, you should be able to confirm by running the lint check locally. There’s even a fix script.

**Comment by seriouslysean** [comment]
@mdrokz looks like I closed it by accident, sorry. On vacation and doing a lot from my phone. 😬

**Comment by mdrokz** [comment]
> @mdrokz yeah, you should be able to confirm by running the lint check locally. There’s even a fix script.

Cool let me do that

**Comment by mdrokz** [comment]
> @mdrokz looks like I closed it by accident, sorry. On vacation and doing a lot from my phone. 😬 

Haha no probs :)

**Comment by mdrokz** [comment]
Just pushed the lint fix

**Review by seriouslysean — APPROVED** [review]
Nice work, thank you!

**Pull request merged**